src/imputors/interp.jl

```diff
@@ -81,10 +69,41 @@ function _impute!(data::AbstractVector{<:Union{T, Missing}}, imp::Interpolate) w
     return data
 end
 
-# Default cases where no rounding behaviour is specified
-_calculate_increment(a, b, n, ::Nothing) = (b - a) / n
-_calculate_increment(a::Unsigned, b::Unsigned, n, r::Nothing) = _calculate_increment(Int(a), Int(b), n, r)
+"""
+Set a vector slice over the values of a generator, starting from `after+1`
+"""
+function _gen_set!(v::AbstractVector, after::Integer, gen)
+    for (i, val) in enumerate(gen)
+       v[after+i] = val
+    end
+end
+
+"""
+Return generator over interpolated values.
+"""
+function _gen_interp(a, b, n, ::Nothing)
+    inc = _calculate_increment(a, b, n)
+    (a + inc*i for i=1:n)
+end
+
+function _gen_interp(a, b, n, r::RoundingMode)
+    inc = _calculate_increment(a, b, n)
+    (round(a + inc*i, r) for i=1:n)
+end
+
+function _gen_interp(a::T, b::T, n, ::Nothing) where {T<:Integer}
+    inc = _calculate_increment(a, b, n)
+    (convert(T, a + inc*i) for i=1:n)
```

Comment:
This isn't necessary because `convert` is called in `setindex!` for us.

https://github.com/JuliaLang/julia/blob/master/base/array.jl#L987